sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/CodeGenInternal/PatchModels/ServicePrincipalInKVParamPatch.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.AI.MetricsAdvisor.Models
+{
+    internal partial class ServicePrincipalInKVParamPatch
+    {
+        // Full qualification must be used so CodeGen skips the generation of this method.
+        // See: https://github.com/Azure/autorest.csharp/issues/793
+        void global::Azure.Core.IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+            writer.WriteNullStringValue("keyVaultEndpoint", KeyVaultEndpoint);
+            writer.WriteNullStringValue("keyVaultClientId", KeyVaultClientId);
+            if (Optional.IsDefined(KeyVaultClientSecret))
```

Comment:
Same reasoning as in other PRs. The service never returns credentials to us, so they would always be sent as null if we treated it as other properties.